src/Raven.Server/Commercial/LetsEncrypt/ZipFileHelper.cs

```diff
@@ -0,0 +1,628 @@
+            Environment.NewLine +
+            "You can do that via: Tools > Options > Advanced > 'Certificates: View Certificates'." +
+            Environment.NewLine;
+
+        if (PlatformDetails.RunningOnPosix)
+            str +=
+                "In Linux, importing the client certificate to the browser might fail for 'Unknown Reasons'." +
+                Environment.NewLine +
+                "If you encounter this bug, use the RavenCli command 'generateClientCert' to create a new certificate with a password." +
+                Environment.NewLine +
+                "For more information on this workaround, read the security documentation in 'ravendb.net'." +
+                Environment.NewLine;
+
+        str +=
+            Environment.NewLine +
+            "It is recommended to generate additional certificates with reduced access rights for applications and users." +
+            Environment.NewLine +
+            "This can be done using the RavenDB Studio, in the 'Manage Server' > 'Certificates' page." +
+            Environment.NewLine;
+
+        if (isCluster)
+        {
+            str +=
+                Environment.NewLine +
+                "You are setting up a cluster. The cluster topology and node addresses have already been configured." +
+                Environment.NewLine +
+                "The next step is to download a new RavenDB server for each of the other nodes." +
+                Environment.NewLine +
+                Environment.NewLine +
+                "When you enter the setup wizard on a new node, please choose 'Continue Existing Cluster Setup'." +
+                Environment.NewLine +
+                "Do not try to start a new setup process again in this new node, it is not supported." +
+                Environment.NewLine +
+                "You will be asked to upload the zip file which was just downloaded." +
+                Environment.NewLine +
+                "The new server node will join the already existing cluster." +
+                Environment.NewLine +
+                Environment.NewLine +
+                "When the wizard is done and the new node was restarted, the cluster will automatically detect it. " +
+                Environment.NewLine +
+                "There is no need to manually add it again from the studio. Simply access the 'Cluster' view and " +
+                Environment.NewLine +
+                "observe the topology being updated." +
+                Environment.NewLine;
+        }
+
+        return str;
+    }
+    
+    public class CompleteAuthorizationAndGetCertificateParameters
+    {
+        public Action OnValidationSuccessful;
+        public SetupInfo SetupInfo;
+        public LetsEncryptClient Client;
+        public (string Challange, LetsEncryptClient.CachedCertificateResult Cache) ChallengeResult;
+        public CancellationToken Token;
+        public RSA ExistingPrivateKey;
+    }
+    
+    public static bool IsValidEmail(string email)
```

Comment:
It's not responsibility of that helper. You can move where it would suit better (maybe to `LetsEncryptUtils` ?) or create a dedicated static class like e.g. `SetupValidator` or `LetsEncryptSetupValidator` - not sure if this is validation specific to let's encrypt setup mode or not